cylc/flow/run_modes/simulation.py

```diff
@@ -266,13 +308,19 @@ def sim_time_check(
     now = time()
     sim_task_state_changed: bool = False
     for itask in itasks:
-        if itask.state.status != TASK_STATUS_RUNNING:
+        if (
+            itask.state.status != TASK_STATUS_RUNNING
+            or itask.tdef.run_mode != RunMode.SIMULATION
+        ):
             continue
```

Comment:
I did wonder about this. Will investigate.

Reply:
Have put a check for the workflow mode, but not the more complex caching of the list of tasks in sim mode, because we're not allowing task level setting of sim mode.